ui/jobs/bar.d.ts

```diff
@@ -0,0 +1,61 @@
+import { Job } from '../../types/job';
+import { JobDetail } from '../../types/event';
+import { ComboCallback } from './combo_tracker';
+
+export interface TimerBar {
+  duration: number;
+}
+
+export interface Box extends HTMLElement {
+  elapsed: number;
+  duration: number;
+  threshold: number;
+  valuescale: number;
+  fg: string;
+  // parentNode: { classList: ()};
+}
+
+type Selector = {
+  id?: string;
+  fgColor?: string;
+  classList?: string[];
+  threshold?: number;
+};
+
+interface ResourceBox extends HTMLDivElement {
+  parentNode: {
+    classList: {
+      add: (s: string) => void;
+      remove: (s: string) => void;
+    };
+  };
+}
+
+export interface Bars {
+  job: Job;
+  speedBuffs: {
+    circleOfPower: boolean;
+    presenceOfMind: boolean;
+    paeonStacks: number;
+    lightningStacks: number;
+    huton: boolean;
+    shifu: boolean;
+    museStacks: 1 | 2 | 3 | 4;
+  };
+  level: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20 |
```

Comment:
What does having a large union type for level get you over just `number`?

Reply:
bypass `kLevelMod[bars.level]`

Reply:
why not just
```typescript
  level: Exclude<Partial<typeof kLevelMod>['length'], (typeof kLevelMod)['length']>;
```

Reply:
I think this will end up being more code to turn the number` you get from OverlayPlugin into this type then it would be to add `?? 0` to looking up something in `kLevelMod`, for what it's worth.